src/ui/ui_settings/__tests__/lib/call_cluster_stub.js

```diff
@@ -0,0 +1,48 @@
+import sinon from 'sinon';
+import expect from 'expect.js';
+import { merge } from 'lodash';
+
+export function createCallClusterStub(index, type, id, esDocSource) {
+  const currentEsDocValue = merge({}, esDocSource);
+
+  const callCluster = sinon.spy(async (method, params) => {
+    expect(params)
```

Comment:
> but do you know if the sinon.stub() that has multiple withArgs() branches will throw if calls don't match the stubbed arg tests?

Off the top of my head, it won't throw and just return `undefined`. That should work okay in most of the cases since we usually test what we expect to see (e.g. verify that expected methods have been called with the right args, correct number of times, in the right order etc.) and not what "the thing" does internally.

> Also, I didn't realize you could do that you did with sinon.match()...

Yeah, it's a pretty powerful thing that validates args and their types in one go.

> I'm all for less stub logic but think that implementing a function inside of sinon.spy() can be a bit easier to read and understand (especially since I don't understand sinon inside an out)

I'm fine with it, whatever you like more